packages/api/src/models/animalBatchModel.js

```diff
@@ -26,6 +27,30 @@ class AnimalBatchModel extends baseModel {
     return 'id';
   }
 
+  static get stringProperties() {
+    return Object.entries(this.jsonSchema.properties)
+      .filter(([, value]) => value.type.includes('string'))
+      .map(([key]) => key);
```

Comment:
Sorry! It seems better to prioritize readability, so no changes are necessary!

My suggestion was to make the array in one loop instead of two.
I know `reduce()` and looping an object keys (for in) are not great in terms of performance, but I can't find good sources...
I learned data structure from a Udemy course, but I will share other resources when I find!

If I were to do it in one loop, I would do:
```
const stringProperties = [];
for (const [key, value] of Object.entries(this.jsonSchema.properties)) {
    if (value.type.includes('string')) {
        stringProperties.push(key);
    }
}
```

Reply:
I like your version the best!

I'm happy to hear reduce is not good for performance because in my opinion it's also not nice for readability 😂